apps/web/src/state/farms/hooks.ts

```diff
@@ -63,6 +71,86 @@ export function useFarmV2PublicAPI() {
   })
 }
 
+export const usePollFarmsAvgInfo = (activeFarms: (V3FarmWithoutStakedValue | V2FarmWithoutStakedValue)[]) => {
+  const { chainId } = useAccountActiveChain()
+  const prevActiveFarms = usePreviousValue(activeFarms)
+
+  const { refetch } = useQuery({
+    queryKey: ['farmsAvgInfo', chainId],
+
+    queryFn: async () => {
+      if (!chainId) return undefined
+      const client = v3Clients[chainId]
+      if (!client) {
+        console.error('[Failed] Trading volume', chainId)
+        return {
+          volumeUSD: 0,
+          tvlUSD: 0,
+          feeUSD: 0,
+        }
+      }
+
+      const addresses = activeFarms
+        .map((farm) => farm.lpAddress)
+        .map((lpAddress) => {
+          return lpAddress.toLowerCase()
+        })
+
+      const rawResult: any | undefined = await multiQuery(
+        (subqueries) => gql`
+      query getVolume {
+        ${subqueries}
+      }
+    `,
+        addresses.map(
+          (tokenAddress) => `
+          t${tokenAddress}:poolDayDatas(first: 7, orderBy: date, orderDirection: desc, where: { pool: "${tokenAddress.toLowerCase()}"}) {
+            volumeUSD
+            tvlUSD
+            feesUSD
+            protocolFeesUSD
+          }
+    `,
+        ),
+        client,
+      )
+
+      const results = mapKeys(rawResult, (_, key) => {
+        return key.substring(1, key.length)
+      })
+
+      return mapValues(results, (value) => {
+        const volumes = value.map((d: { volumeUSD: string }) => Number(d.volumeUSD))
+        const feeUSDs = value.map(
+          (d: { feesUSD: string; protocolFeesUSD: string }) => Number(d.feesUSD) - Number(d.protocolFeesUSD),
+        )
+        return {
+          volumeUSD: averageArray(volumes),
+          tvlUSD: parseFloat(value[0]?.tvlUSD) || 0,
+          feeUSD: averageArray(feeUSDs),
+        }
+      })
+    },
+
+    enabled: Boolean(chainId && activeFarms),
+    refetchOnReconnect: false,
+    refetchOnWindowFocus: false,
+    refetchOnMount: false,
+  })
+
+  useEffect(() => {
+    if (activeFarms) {
+      if (prevActiveFarms) {
+        if (activeFarms.length !== prevActiveFarms.length) {
+          refetch()
+        }
+      } else {
+        refetch()
+      }
+    }
+  }, [refetch, activeFarms, prevActiveFarms])
```

Comment:
using length to compare doesn't look ideal. Can we use active farm addresses as query key?

Reply:
is there any better way to deal with this? Tow concerns on my end
1. This approach is gonna have unexpected result if the new active farm list has the same length as the previous one

2. Not sure if separating the poll query and the consumer query is a good practice. It'll make it harder to reuse hooks because they have some dependency relationship covered inside the implementation. For example, if I don't check the code of `usePoolAvgInfo`, I won't be able to understand that `useFarmCache` is only valid when `usePollFarmsAvgInfo` is used. It creates a hidden dependency relationship between `usePoolAvgInfo` -> `usePollFarmsAvgInfo`

Reply:
Can we have a hook called `useActiveFarms` and reuse that in the `usePoolAvgInfo` hook?

Reply:
I'm thinking relationship like
```typescript
function useActiveFarms(chainId) {}
function useFarmsInfo(activeFarms) {}
function usePoolAvgInfo() {
  const activeFarms = useActiveFarms()
  const farmsInfo = useFarmsInfo(activeFarms)
}
```

Reply:
by changing a lot you mean the reference of the farm list is changing? Can we just use the address of the farm list as query key instead?